paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -44,23 +42,36 @@ internal class CustomerSheetViewModel @Inject constructor(
         }
     }
 
+    fun providePaymentMethodName(code: PaymentMethodCode?): String {
+        val paymentMethod = lpmRepository.fromCode(code)
+        return paymentMethod?.displayNameResource?.let {
+            resources.getString(it)
+        }.orEmpty()
+    }
+
     private fun loadPaymentMethods() {
         viewModelScope.launch {
-            var savedPaymentMethods: List<PaymentOptionsItem.SavedPaymentMethod> = emptyList()
-            var selectedPaymentMethodId: String? = null
+            var savedPaymentMethods: List<PaymentMethod> = emptyList()
+            var paymentSelection: PaymentSelection? = null
             var errorMessage: String? = null
             customerAdapter.retrievePaymentMethods().fold(
                 onSuccess = { paymentMethods ->
-                    savedPaymentMethods = paymentMethods.map { paymentMethod ->
-                        PaymentOptionsItem.SavedPaymentMethod(
-                            displayName = paymentMethod
-                                .getLabel(resources)
-                                .toString(),
-                            paymentMethod = paymentMethod
-                        )
-                    }
-                    customerAdapter.retrieveSelectedPaymentOption().onSuccess { paymentMethod ->
-                        selectedPaymentMethodId = paymentMethod?.id
+                    savedPaymentMethods = paymentMethods
+                    customerAdapter.retrieveSelectedPaymentOption().onSuccess { paymentOption ->
+                        paymentSelection = when (paymentOption) {
+                            is CustomerAdapter.PaymentOption.GooglePay -> {
```

Comment:
Should we map these in `CustomerAdapter.PaymentOption`?

Reply:
Will address in the next PR!